pkg/ingester/ingester.go

```diff
@@ -1535,8 +1533,9 @@ func (i *Ingester) AllUserStats(_ context.Context, req *client.UserStatsRequest)
 // So, 1 MB limit will prevent reaching the limit and won't affect performance significantly.
 const labelNamesAndValuesTargetSizeBytes = 1 * 1024 * 1024
 
-func (i *Ingester) LabelNamesAndValues(request *client.LabelNamesAndValuesRequest, stream client.Ingester_LabelNamesAndValuesServer) error {
-	if err := i.checkRunning(); err != nil {
+func (i *Ingester) LabelNamesAndValues(request *client.LabelNamesAndValuesRequest, stream client.Ingester_LabelNamesAndValuesServer) (lnvErr error) {
+	defer func() { lnvErr = i.handleReadError(lnvErr) }()
```

Comment:
Same comment as before, i.e. I think that the returned error should be named `err` to avoid confusion and potential programming mistakes.